spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/VectorUtils.java

```diff
@@ -20,24 +20,22 @@
 import org.apache.commons.lang.ArrayUtils;
 import org.apache.spark.mllib.linalg.SparseVector;
 
-import java.util.ArrayList;
-import java.util.Arrays;
 import java.util.List;
 
 /**
  * Vector utility class to handle sparse vectors.
  */
 public class VectorUtils {
-  private static final Schema SPARSE_SCHEMA =
+  public static final Schema SPARSE_SCHEMA =
     Schema.recordOf("sparseVector", Schema.Field.of("size", Schema.of(Schema.Type.INT)),
                     Schema.Field.of("indices", Schema.arrayOf(Schema.of(Schema.Type.INT))),
                     Schema.Field.of("vectorValues", Schema.arrayOf(Schema.of(Schema.Type.DOUBLE))));
 
   public static StructuredRecord asRecord(SparseVector vector) {
     return StructuredRecord.builder(SPARSE_SCHEMA)
       .set("size", vector.size())
-      .set("indices", new ArrayList<>(Arrays.asList(ArrayUtils.toObject(vector.indices()))))
-      .set("vectorValues", new ArrayList<>(Arrays.asList(ArrayUtils.toObject(vector.values()))))
+      .set("indices", ArrayUtils.toObject(vector.indices()))
```

Comment:
The fix for bug HYDRATOR-1062 ensures that the unit test cases run successfully for StructuredRecord using int[] or double[]. 
However, since RecordConverter still tries to directly type cast value to Object[]  : ``Object[] valuesArr = (Object[]) values;`` ([line #71](https://github.com/caskdata/hydrator-plugins/blob/aa1e4caafe2b603bffe7714125b0e421d22294a4/hydrator-common/src/main/java/co/cask/hydrator/common/RecordConverter.java#L71) ), and the sinks like TPFSAvro internally call the RecordConverter, the pipelines for the feature generator plugins are still failing on UI with ClassClastException.